prometheus/not_compatible.go

```diff
@@ -94,3 +120,71 @@ func (c *notCompatibleHTTPClient) SetLookbackInterval(interval time.Duration) {
 func (c *notCompatibleHTTPClient) GetURL() string {
 	return ""
 }
+
+func uniqueLabelNameAndValues(
+	ctx context.Context,
+	prometheusClient v1.API,
+	start, end time.Time,
+	currMatchers []*labels.Matcher,
+) (map[string]map[string]struct{}, error) {
+	var (
+		metricName   string
+		metricLabels []*labels.Matcher
+	)
+	for _, matcher := range currMatchers {
+		if matcher.Name == model.MetricNameLabel {
+			metricName = string(matcher.Value)
+		} else {
+			metricLabels = append(metricLabels, matcher)
+		}
+	}
+
+	var match strings.Builder
+	if _, err := match.WriteString(metricName); err != nil {
+		return nil, err
+	}
+	if len(metricLabels) > 0 {
+		if _, err := match.WriteString("{"); err != nil {
+			return nil, err
+		}
+		for i, matcher := range metricLabels {
+			if i != 0 {
+				if _, err := match.WriteString(","); err != nil {
+					return nil, err
+				}
+			}
+			if _, err := match.WriteString(matcher.String()); err != nil {
+				return nil, err
+			}
+		}
+		if _, err := match.WriteString("}"); err != nil {
+			return nil, err
+		}
+	}
+
+	results, _, err := prometheusClient.Series(ctx, []string{match.String()},
+		start, end)
+	if err != nil {
+		return nil, err
+	}
+
+	// deduplicated is a de-duplicated result set.
+	deduplicated := make(map[string]map[string]struct{})
```

Comment:
In cases when we only want to know the values of one label, computing all the other stuff seems like busywork.